compositor_common/src/error.rs

```diff
@@ -44,6 +44,58 @@ impl Display for UnusedNodesError {
     }
 }
 
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum ConstraintsValidationError {
+    #[error("Specification of node \"{1}\" doesn't satisfy node input pads count constraint.")]
+    InvalidInputsPads(#[source] InputsCountValidationError, NodeId),
+}
+
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum InputsCountValidationError {
+    #[error(
+        "Nodes that use transformation \"{}\" doesn't expect input pads, but {} input pads were specified.",
+        transformation_name,
+        defined_input_pads_count
+    )]
```

Comment:
```suggestion
    #[error("Nodes that use transformation \"{transformation_name}\" don't support any inputs.")]
```
Also, "Nodes that use transformation" makes sense for builtins, but wouldn't e.g. for image renderer